pkgs/by-name/si/signal-desktop-source/package.nix

```diff
@@ -145,12 +146,28 @@ stdenv.mkDerivation (finalAttrs: {
       die "libsignal-client version mismatch"
     fi
 
-    if [ "`jq -r '.dependencies."@signalapp/ringrtc"' < package.json`" != "${ringrtc-bin.version}" ]
+    if [ "`jq -r '.dependencies."@signalapp/ringrtc"' < package.json`" != "${ringrtc.version}" ]
     then
       die "ringrtc version mismatch"
     fi
 
-    cp -r ${ringrtc-bin} node_modules/@signalapp/ringrtc/build
+    mkdir -p node_modules/@signalapp/ringrtc/build
+    install -D ${ringrtc}/lib/libringrtc${stdenv.hostPlatform.extensions.library} node_modules/@signalapp/ringrtc/build/${
+      {
+        "linux" = "linux";
+        "darwin" = "darwin";
+      }
+      .${stdenv.hostPlatform.parsed.kernel.name}
+        or (throw "unsupported platform ${stdenv.hostPlatform.parsed.kernel.name}")
+    }/libringrtc-${
+      {
+        # https://nodejs.org/api/os.html#osarch
+        "x86_64" = "x64";
+        "aarch64" = "arm64";
+      }
+      .${stdenv.hostPlatform.parsed.cpu.name}
+        or (throw "unsupported platform ${stdenv.hostPlatform.parsed.cpu.name}")
+    }.node
```

Comment:
Surely this logic would look better in a `let in` block...

Or maybe use `*` instead of being explicit with the dirs?